test/runtests.jl

```diff
@@ -646,3 +646,8 @@ end
     str = String(take!(io))
     @test str == "StructArray(::Array{Int64,1}, ::Array{Int64,1})"
 end
+
+@testset "broadcast" begin
+    s = StructArray{ComplexF64}((rand(2,2), rand(2,2)))
+    @test isa(s .+ s, StructArray)
```

Comment:
Probably this also needs to test that things like `broadcast(t -> 1, s)` don't give an error (I'm not sure what exactly should happen when the result of broadcasting StructArrays has a eltype that is not a struct).